dns-policy-operator/src/dns_policy.py

```diff
@@ -0,0 +1,250 @@
+                f"Reason: {e}. Output: {e.output}"
+            )
+            logger.exception(error_msg)
+            raise InstallError(error_msg) from e
+
+    def configure(self, config: dict[str, str]) -> None:
+        """Configure the dns-policy service.
+
+        Args:
+            config: dict of configuration values
+
+        Raises:
+            ConfigureError: when encountering a SnapError
+        """
+        try:
+            cache = snap.SnapCache()
+            charmed_bind = cache[constants.DNS_SNAP_NAME]
+            charmed_bind.set(config)
+        except snap.SnapError as e:
+            error_msg = (
+                f"An exception occurred when configuring {constants.DNS_SNAP_NAME}. Reason: {e}"
+            )
+            logger.error(error_msg)
+            raise ConfigureError(error_msg) from e
+
+    def command(self, cmd: str) -> str:
+        """Run manage command of the dns-policy service.
+
+        Args:
+            cmd: command to execute by django's manage script
+            env: environment
+
+        Raises:
+            CommandError if the command call errors
+
+        Returns:
+            The resulting output of the command's execution
+        """
+        try:
+            # We ignore security issues with this subprocess call
+            # as it can only be done from the operator of the charm
+            return subprocess.check_output(
+                ["sudo", "snap", "run", f"{constants.DNS_SNAP_NAME}.manage"] + cmd.split(),
+            ).decode(  # nosec
+                "utf-8"
+            )
+        except subprocess.SubprocessError as e:
+            raise CommandError(str(e)) from e
+
+    def get_api_root_token(self) -> str:
+        """Get API root token."""
+        try:
+            res = self.command("get_root_token")
+            tokens = json.loads(res)
+        except json.decoder.JSONDecodeError as e:
+            raise RootTokenError(str(e)) from e
+        if not isinstance(tokens, dict) or "access" not in tokens or tokens["access"] == "":
+            raise RootTokenError("Invalid root token!")
+        return tokens["access"]
+
+    def send_requests(self, token: str, record_requests: list[RequirerEntry]) -> bool:
+        """Send record requests.
+
+        Args:
+            token: root token for the API
+            record_requests: list of record requests from the relations
+
+        Returns:
+            True if the request succeeded False otherwise
+
+        Raises:
+            ApiError if the request errors
+        """
+        try:
+            req = requests.post(
+                f"{constants.DNS_POLICY_ENDPOINTS_BASE}/",
+                headers={
+                    "Content-Type": "application/json",
+                    "Authorization": f"Bearer {token}",
+                },
+                timeout=10,
+                data=json.dumps([x.model_dump() for x in record_requests]),
+            )
```

Comment:
IMO a non 20X HTTP code is an error condition and should be handled as an exception. Hence, I think you should `raise_for_status` instead of returning

Reply:
Changed to use raise_for_status